go/src/meguca/parser/commands.go

```diff
@@ -55,14 +55,67 @@ func parseCommand(match []byte, board string, ip string) (com common.Command, er
 	case bytes.Equal(match, []byte("pyu")):
 		if boardConfig.Pyu {
 			com.Type = common.Pyu
-			com.Pyu, err = db.IncrementPcount(boardConfig.ID)
+			now := time.Now()
+			exists, err := db.PyuLimitExists(ip, board)
+
+			if err != nil {
+				return com, err
+			}
+
+			if exists == 0 {
+				err = db.WritePyuLimit(ip, board, now.Add(-time.Second))
+
+				if err != nil {
+					return com, err
+				}
+			}
+
+			limit, err := db.GetPyuLimit(ip, board)
+
+			if err != nil {
+				return com, err
+			}
+			
+			expires, err := db.GetPyuLimitExpires(ip, board)
+
+			if err != nil {
+				return com, err
+			}
+
+			if limit == 0 && expires.Before(now) {
+				limit = 4
+				err = db.ResetPyuLimit(ip, board)
```

Comment:
The time seems to not evaluate correctly, even though in `println`, `now` and `expires` are both accurate to what they should be.
I'm not sure what to do, it causes this if statement to not work correctly.

Reply:
They are now.
Still no way to get it to work.

Reply:
Can you paste what this outputs: `fmt.Printf("%d, %#v, %#v\n", limit, now, expires)`? Also, you should really use a debugger for this. Even gdb is better than nothing and I strongly recommend some text editor or IDE with Go debugger support.

Reply:
`4, time.Time{wall:0x11495ce9, ext:63660713399, loc:(*time.Location)(nil)}, time.Time{wall:0x11495e88, ext:63660713398, loc:(*time.Location)(nil)}`
`3, time.Time{wall:0x2523b0c7, ext:63660713402, loc:(*time.Location)(nil)}, time.Time{wall:0x11495e88, ext:63660713398, loc:(*time.Location)(nil)}`
`2, time.Time{wall:0x326796fc, ext:63660713404, loc:(*time.Location)(nil)}, time.Time{wall:0x11495e88, ext:63660713398, loc:(*time.Location)(nil)}`
`1, time.Time{wall:0xcdaaf90, ext:63660713407, loc:(*time.Location)(nil)}, time.Time{wall:0x11495e88, ext:63660713398, loc:(*time.Location)(nil)}`
`4, time.Time{wall:0x171d2e91, ext:63660713410, loc:(*time.Location)(nil)}, time.Time{wall:0x11495e88, ext:63660713398, loc:(*time.Location)(nil)}`

>debugger
>IDE

Are there even any decent Go IDEs with Linux support?